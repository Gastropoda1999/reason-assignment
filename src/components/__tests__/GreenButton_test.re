open Jest;
open Expect;

let _ =

describe("GreenButton", () => {
  let action = Action.Add;
  let dispatch = _action => ();

  test("renders", () => {
    let tree = <GreenButton action dispatch />
      |> ReactShallowRenderer.renderWithRenderer;

    expect(tree) |> toMatchSnapshot;
  });

  test("renders a class name", () => {
    let className = Styles.make(~color="rebeccapurple", ()) |> Styles.className;
    let tree = <GreenButton action className dispatch />
      |> ReactShallowRenderer.renderWithRenderer;

    expect(tree) |> toMatchSnapshot;
  });
});
