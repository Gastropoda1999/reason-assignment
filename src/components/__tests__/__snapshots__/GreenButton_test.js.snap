// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreenButton renders 1`] = `
<OperationButton
  reasonProps={
    Array [
      Array [
        "OperationButton",
        [Function],
        Array [
          0,
        ],
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
        0,
        [Function],
        0,
      ],
    ]
  }
/>
`;

exports[`GreenButton renders a class name 1`] = `
<OperationButton
  reasonProps={
    Array [
      Array [
        "OperationButton",
        [Function],
        Array [
          0,
        ],
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
        0,
        [Function],
        0,
      ],
    ]
  }
/>
`;
